IMA-WorldHealth/bhima
# Add UI to change the order of the cost center step-down allocation steps

This PR adds a modal that lets the user change the order of the step-down steps.

NOTE:  I believe this is working correctly.  However, when the order of the steps are changed, the cost center step-down report is broken:  It performs an allocation on one of the principal centers before finishing all auxiliary centers.   I suggest releasing this PR and then addressing the report bug separately.

Closes https://github.com/IMA-WorldHealth/bhima/issues/5984

**TESTING**
- For basic testing, use bhima_test
- Reload the bhima_test db and the sample data
- Try: Cost Centers > Menu > Edit step order... 
   - Use the up/down arrows to reorder the steps
   - Click [Submit]
   - Re-open the step-order editor to verify the change
- Try: Cost Centers > Report > Cost Center Allocations
   - Note the auxiliary center order has changed
   - Note that the bug: a principal center is being processed before all the auxiliary centers are processed.
- Try it with a production db
